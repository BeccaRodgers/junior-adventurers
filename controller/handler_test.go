package controller

import (
	"bytes"
	"context"
	approvals "github.com/approvals/go-approval-tests"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
	"github.com/yosssi/gohtml"
	"junior-adventurers/fixtures"
	"junior-adventurers/memdb"
	"net/http"
	"net/http/httptest"
	"testing"
)

func TestHandler_Homepage(t *testing.T) {
	approveGetPage(t, "/")
}

func TestHandler_GuildPage(t *testing.T) {
	approveGetPage(t, "/guilds/1")
}

func TestHandler_NewMemberPage(t *testing.T) {
	approveGetPage(t, "/members")
}

func approveGetPage(t *testing.T, path string) {
	t.Helper()

	guilds := memdb.NewGuildRepository()
	require.NoError(t, fixtures.InsertGuilds(context.TODO(), guilds))
	members := memdb.NewMemberRepository()
	require.NoError(t, fixtures.InsertMembers(context.TODO(), members))

	handler := Handler(guilds, members)
	request := httptest.NewRequest(http.MethodGet, path, nil)
	recorder := httptest.NewRecorder()
	handler.ServeHTTP(recorder, request)
	assert.Equal(t, http.StatusOK, recorder.Code)
	bodyBytes := gohtml.FormatBytes(recorder.Body.Bytes())
	var formattedOutput bytes.Buffer
	formattedOutput.Write(bodyBytes)
	approvals.Verify(t, &formattedOutput)
}
