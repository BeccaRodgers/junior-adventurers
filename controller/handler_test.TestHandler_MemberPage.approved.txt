<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>
      Junior Adventurers
    </title>
    <link rel="stylesheet" href="/static/style.min.css">
    <link rel="icon" type="image/x-icon" href="/static/favicon.ico">
    <script src="/static/htmx.min.js"></script>
  </head>
  <body hx-target="this">
    <section class="section">
      <div class="container has-text-centered">
        <h1 class="title">
          Angela
        </h1>
      </div>
      <br>
      <div class="container">
        <p class="block">
          <b>
            Member ID:
          </b>
          1
        </p>
        <p class="block">
          <b>
            Age:
          </b>
          10
        </p>
        <p class="block">
          <b>
            Species:
          </b>
          Human
        </p>
        <button class="button is-link" hx-get="/" hx-push-url="true">
          Home
        </button>
      </div>
    </section>
  </body>
</html>